portfolio/src/main/webapp/script.js

```diff
@@ -144,28 +144,51 @@ function createListElement(comment, nickname) {
   let h3Element = document.createElement('h3');
   let h5Element = document.createElement('h5');
   let pElement = document.createElement('p');
+  let spanElement = document.createElement('span');
   // Fill elements
   h3Element.innerText = comment.title;
   let dateTime = comment.postTime.split(" ");
   let date = dateTime.slice(0, 3).join(" ");
   let time = dateTime.slice(3, ).join(" ");
   h5Element.innerText = "Posted on " + date + " at " + time;
-  h5Element.innerText += "\nPosted by " + nickname;
+  if (g_currentUserId === comment.userId){
+    h5Element.innerText += "\nPosted by You as " + nickname;
+  } else {
+    h5Element.innerText += "\nPosted by " + nickname;
+  }
   pElement.innerText = comment.message;
+  spanElement.setAttribute("onclick", "deleteComment('" + comment.id + "')");
+  spanElement.setAttribute("class", "fa fa-trash-o fa-2x");
+  spanElement.setAttribute("aria-hidden", "true");
   // Add elements to the li element
   liElement.appendChild(h3Element);
   liElement.appendChild(h5Element);
   liElement.appendChild(pElement);
+  if (g_currentUserId === comment.userId || g_isAdmin === "true"){
+    liElement.appendChild(spanElement);
+  }
   return liElement;
 }
 
-// Fetches to delete all comments
+// Deletes a single comment
+function deleteComment(commentId) {
+  let parameters = new URLSearchParams('id=' + commentId);
+  console.log('/delete-some-data', parameters);
+  fetch('/delete-some-data', {method: "post", body: parameters});
+  getComments();
```

Comment:
`fetch` here is an asynchronous call and it's not guaranteed that the delete servlet would complete before `getComments` is invoked.  Meaning you can run into some scenarios (with really slow internet maybe) where getComments would return the previous set including the deleted comment.

It's good as is since your delete servlet is pretty light weight and would return quick.  There's a couple options though if you would like to improve upon this.